src/components/app/list-new/GenericAppList.tsx

```diff
@@ -93,36 +128,53 @@ export default function ExternalArgoList({
         if (dataStateType == AppListViewType.LIST) {
             handleFilteration()
         }
-    }, [argoAppsList])
+    }, [appsList])
 
     useEffect(() => {
         // fetch external apps list
-        updateDataSyncing(true)
-        setDataStateType(AppListViewType.LOADING)
-        getArgoInstalledExternalApps(clusterIdsCsv, appStatus)
-            .then((argoAppsListResponse) => {
-                setArgoAppsList(argoAppsListResponse.result)
-            })
-            .catch((errors: ServerErrors) => {
-                showError(errors)
-                setDataStateType(AppListViewType.ERROR)
-                setErrorResponseCode(errors.code)
-            })
-            .finally(() => {
-                updateDataSyncing(false)
-                setFetchingExternalAppsState(false)
-                setDataStateType(AppListViewType.LIST)
-            })
+        if (isArgoCDAppList) {
+            setDataStateType(AppListViewType.LOADING)
```

Comment:
Can convert to two methods for argo and flux and do early returns instead of else if